habitat_baselines/rl/models/rnn_state_encoder.py

```diff
@@ -4,169 +4,201 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Tuple
+from typing import Dict, Tuple
 
+import numpy as np
 import torch
 import torch.nn as nn
 from torch.nn.utils.rnn import PackedSequence
 
+from habitat_baselines.common.tensor_dict import TensorDict
+
 
 def _invert_permutation(permutation: torch.Tensor) -> torch.Tensor:
-    output = torch.empty_like(permutation)
+    output = torch.empty_like(permutation.view(-1))
     output.scatter_(
         0,
-        permutation,
+        permutation.view(-1),
         torch.arange(0, permutation.numel(), device=permutation.device),
     )
-    return output
+    return output.view_as(permutation)
+
 
+def _np_invert_permutation(permutation: np.ndarray) -> np.ndarray:
```

Comment:
this private method does not seem to be used.

Reply:
It will be.